src/main/java/com/istb/app/services/profesor/ProfesorServiceImpl.java

```diff
@@ -137,20 +147,31 @@ public void delete(int id) {
 
 		Role rol = rolService.findByNombre("Docente");
 		profesor.getUsuario().addRol(rol);
-		
-		
-		
+		String token = generateToken();
+		profesor.setTokenVerification(token);
 
 		userService.save(profesor.getUsuario()).forEach((k, v) -> errorAttributes.put(k, v));
 
 		if (!errorAttributes.isEmpty()) {
 			return errorAttributes;
 		}
 
-		profesorRepository.save(profesor);
-		
+		Profesor profesorGuarded = profesorRepository.save(profesor);
+
+		Map<String, Object> data = new HashMap<>();
+		data.put("profesor", profesorGuarded.getId());
```

Comment:
Deberías ponerle `profesorId` a la clave, para que en la plantilla no existan ambiguedades. Cuando ves el HTML;

`<a th:href="${url+'/verify/'+profesor+'/'+token}"`

la primera pregunta es ¿Que es `profesor`?, luego visitas este archivo y queda claro. Debes evitar, cuanto se pueda, que para entender una parte del código alguien viaje a otro archivo.

```suggestion
		data.put("profesorId", profesorGuarded.getId());
```
Asi el HTML queda mejor descrito: `<a th:href="${url+'/verify/'+profesorId+'/'+token}"`